src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -607,78 +604,84 @@ private void PopulateSourceRepoList(string optionalSelectSourceName = null)
         /// <summary>
         /// This method is called from several event handlers. So, consolidating the use of JTF.Run in this method
         /// </summary>
-        internal void SearchPackagesAndRefreshUpdateCount(string searchText, bool useCacheForUpdates)
+        internal void SearchPackagesAndRefreshUpdateCount(bool useCacheForUpdates)
         {
-            SearchPackagesAndRefreshUpdateCount(searchText: searchText, useCacheForUpdates: useCacheForUpdates, pSearchCallback: null, searchTask: null);
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                await SearchPackagesAndRefreshUpdateCountAsync(
+                searchText: _windowSearchHost.SearchQuery.SearchString,
+                useCacheForUpdates: useCacheForUpdates,
+                pSearchCallback: null,
+                searchTask: null);
+            });
         }
 
         /// <summary>
         /// This method is called from several event handlers. So, consolidating the use of JTF.Run in this method
         /// </summary>
-        internal void SearchPackagesAndRefreshUpdateCount(string searchText, bool useCacheForUpdates, IVsSearchCallback pSearchCallback, IVsSearchTask searchTask)
+        internal async Task SearchPackagesAndRefreshUpdateCountAsync(string searchText, bool useCacheForUpdates, IVsSearchCallback pSearchCallback, IVsSearchTask searchTask)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
-            {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            VSThreadHelper.ThrowIfNotOnUIThread();
 
```

Comment:
This violates the first threading rule, and is caught by [VSTHRD109](https://github.com/Microsoft/vs-threading/blob/master/doc/analyzers/VSTHRD109.md) (a relatively new analyzer that may not be in the version you're using).

Reply:
This was the internal API and we were already making sure to always call it on UI thread. But I get your point and will update it.